### DISCUSSION TITLE
sigma coordinates

### DISCUSSION THREAD

**Discussion by ana-machado** [Parcels-code/Parcels#1021] [General]
Hello everyone,
Can anyone update me on the status of the implementation of sigma coordinates in parcels? And do you know of any studies successfully using parcels with ROMS outputs? I have some scripts from last summer, to study larval dispersal with ROMS outpus, and I am now getting back to them (finally have the ROMS simulations) and would like to know if there were any advances. I know someone was working on implementing sigma coordinates varying in time. Is this done yet? 
Thank you very much :)
Ana

**Comment by reint-fischer** [comment]
Hi @ana-machado ,

I am not working with ROMS myself, so I am not sure about the exact compatibility at the moment. To confirm, last spring this (#765) is where the discussion was left? I know @DEHewitt was working on some simulations recently, is that working for you? I also know that Parcels supports time-varying grids through [this implementation](https://github.com/OceanParcels/parcels/blob/master/parcels/examples/example_dask_chunk_OCMs.py#L96-L129), although I don't know how that works with a spherical mesh and the [grid indexing used in ROMS](https://github.com/OceanParcels/parcels/discussions/1013#discussion-3311396). I believe @cjongedijk was working with the time-varying grids: are those sigma grids Cleo?

**Comment by DEHewitt** [comment]
Hi @ana-machado,

@HaydenSchilling and myself have been working on using recent ROMS output to model larval dispersal in the east Australian region. We have been attempting 3d-advection which has been made tricky given the sigma-coordinates, but we have got a combination of #765 and a buoyancy kernel we wrote to do what we wanted (although it is _very_ hacky). The ROMS model we're using doesn't have vertical velocities (`W`) either.  

@HaydenSchilling and @jaseeverett published a paper last year using just the surface layers of a ROMS model which can be found [here](https://onlinelibrary.wiley.com/doi/full/10.1111/fog.12473?casa_token=98uNlelUatAAAAAA%3AKRDJ7lW_IG4jwc4pnO5ZzgsZl1TZp09Tuqoa5ze3nepRCssPtnpRcmKoou7F15rbasmZ4n_XLd46fhtHGQ), and the associated repo [here](https://github.com/jaseeverett/TailorParticleTracking). If you can justify only using the surface layers this would be much easier, see section 2.1 and Fig. S1 and S2 in [this paper](https://onlinelibrary.wiley.com/doi/10.1111/fog.12213) for an example of what I mean.

I hope this helps!

**Reply by ana-machado** [reply]
Hi @DEHewitt and @reint-fischer 

Thank you so much for your reply. I have also been using #765 together with some buoyancy and diurnal vertical migration kernels that I wrote. It seems to be working ok. I am considering the depth of the sigma coordinates does not change in time, I think it was under development the possibility of considering them varying in time, but it seems there were no advances in this.

I am now trying to add diffusion to my simulations, I was thinking of using AdvectionDiffusionM1 kernels, but just noticed @HaydenSchilling  and @jaseeverett used instead BrownianMotion2D. What are you using in your simulations, any of these? and why?

Thank you again :)
Ana

**Reply by DEHewitt** [reply]
Hi @ana-machado, yeah we've stitched together something simple to get particles to rise at a constant rate. It's a little messy though.

I'm currently using `DiffusionUniformKh`, which I _think_ replaced `BrownianMotion2D`. With this diffusion scheme you add zonal and meridional diffusion constants to the field. @HaydenSchilling and @jaseeverett define the constants on lines 52 and 53, and then add them to the field on lines 89 and 90.
